File: backend/app/core/roles.py
import enum


class Role(str, enum.Enum):
    admin = "admin"
    receptionist = "receptionist"
    doctor = "doctor"
    head_nurse = "head_nurse"
    lab_staff = "lab_staff"
    pharmacist = "pharmacist"
    patient = "patient"


ALL_ROLES = tuple(r.value for r in Role)

File: web/src/lib/roles.ts
export type Role =
  | "admin"
  | "receptionist"
  | "doctor"
  | "head_nurse"
  | "lab_staff"
  | "pharmacist"
  | "patient";

export const ROLE_HOME: Record<Role, string> = {
  admin: "/admin",
  receptionist: "/reception",
  doctor: "/doctor",
  head_nurse: "/nurse",
  lab_staff: "/lab",
  pharmacist: "/pharmacy",
  patient: "/login", // patient role only uses the mobile app
};

export const ROLE_LABEL: Record<Role, string> = {
  admin: "Super Administrator",
  receptionist: "Receptionist",
  doctor: "Doctor",
  head_nurse: "Head Nurse",
  lab_staff: "Lab Staff",
  pharmacist: "Pharmacist",
  patient: "Patient",
};
